var/spack/repos/builtin/packages/mathematica/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Mathematica(Package):
+    """Mathematica: high-powered computation with thousands of Wolfram Language
+       functions, natural language input, real-world data, mobile support.
+
+       Note: A manual download is required for Mathematica.
+       Spack will search your current directory for the download file.
+       Alternatively, add this file to a mirror so that Spack can find it.
+       For instructions on how to set up a mirror, see
+       http://spack.readthedocs.io/en/latest/mirrors.html"""
+
+    homepage = "https://www.wolfram.com/mathematica/"
+    url = 'file://{0}/Mathematica_12.0.0_LINUX.sh'.format(os.getcwd())
```

Comment:
This is not the right way to do this.  If you have a proprietary package, then the steps are:
1. Don't have any URL in the package (or use a dummy URL, if no-URL breaks things).
2. Manually download the tarball and place in the Spack cache.
3. "spack install" as normal.

Please get rid of this "hack" URL.

Reply:
My apologies, I was working using [sspace-standard](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/sspace-standard/package.py) as a reference. I'm unsure what you mean by "dummy URL" -- If I remove the URL or replace it with an empty string, spack refuses to install the package: `==> Error: Package Mathematica has no version with a URL.`